## Assignment 4

In [1]:
import pandas as pd     # importing pandas

In [2]:
df = pd.read_csv("MERGED2017_18_PP.csv")      # reading the csv file from local  
df

C:\Users\priya\AppData\Local\Temp\ipykernel_4468\3895413984.py:1: DtypeWarning: Columns (1729) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv("MERGED2017_18_PP.csv")      # reading the csv file from local


UNITID    OPEID  OPEID6  \
0       100654   100200    1002   
1       100663   105200    1052   
2       100690  2503400   25034   
3       100706   105500    1055   
4       100724   100500    1005   
...        ...      ...     ...   
7107  48312404   108163    1081   
7108  48387801  4223701   42237   
7109  48400201  4228101   42281   
7110  48778201  4221501   42215   
7111  49005401  4182601   41826   

                                                 INSTNM             CITY  \
0                              Alabama A & M University           Normal   
1                   University of Alabama at Birmingham       Birmingham   
2                                    Amridge University       Montgomery   
3                   University of Alabama in Huntsville       Huntsville   
4                              Alabama State University       Montgomery   
...                                                 ...              ...   
7107                   Arizona State University at Yuma             Yuma   
7108  Bay Area Medical Academy - San Jose Satellite ...         San Jose   
7109           High Desert Medical College - Bakerfield      Bakersfield   
7110                                       BCI - Malden           Malden   
7111               HCI College - Fort Lauderdale Campus  Fort Lauderdale   

     STABBR         ZIP  ACCREDAGENCY  INSTURL  NPCURL  ...  \
0        AL       35762           NaN      NaN     NaN  ...   
1        AL  35294-0110           NaN      NaN     NaN  ...   
2        AL  36117-3553           NaN      NaN     NaN  ...   
3        AL       35899           NaN      NaN     NaN  ...   
4        AL  36104-0271           NaN      NaN     NaN  ...   
...     ...         ...           ...      ...     ...  ...   
7107     AZ   853656900           NaN      NaN     NaN  ...   
7108     CA       95113           NaN      NaN     NaN  ...   
7109     CA       93301           NaN      NaN     NaN  ...   
7110     MA   021480000           NaN      NaN     NaN  ...   
7111     FL       33309           NaN      NaN     NaN  ...   

      OMENRUP_PARTTIME_POOLED_SUPP  FTFTPCTPELL  FTFTPCTFLOAN   UG12MN  \
0                              NaN       0.7106        0.7418   5207.0   
1                              NaN       0.4009        0.5488  13308.0   
2                              NaN       0.5000        0.5000    431.0   
3                              NaN       0.2012        0.4231   7519.0   
4                              NaN       0.7594        0.7402   5229.0   
...                            ...          ...           ...      ...   
7107                           NaN          NaN           NaN      NaN   
7108                           NaN          NaN           NaN      NaN   
7109                           NaN          NaN           NaN      NaN   
7110                           NaN          NaN           NaN      NaN   
7111                           NaN          NaN           NaN      NaN   

       G12MN  SCUGFFN  POOLYRS_FTFTAIDPCT  FTFTPCTPELL_POOLED_SUPP  \
0     1185.0   1410.0                 NaN                      NaN   
1     9888.0   1948.0                 NaN                      NaN   
2      437.0      4.0                 NaN                      NaN   
3     2378.0   1203.0                 NaN                      NaN   
4      687.0   1143.0                 NaN                      NaN   
...      ...      ...                 ...                      ...   
7107     NaN      NaN                 NaN                      NaN   
7108     NaN      NaN                 NaN                      NaN   
7109     NaN      NaN                 NaN                      NaN   
7110     NaN      NaN                 NaN                      NaN   
7111     NaN      NaN                 NaN                      NaN   

      FTFTPCTFLOAN_POOLED_SUPP  SCUGFFN_POOLED  
0                          NaN             NaN  
1                          NaN             NaN  
2                          NaN          

# Displaying the number of rows and columns

In [3]:
df.shape                                         #shape of dataframe

(7112, 1986)

There are total 7112 rows and 1986 columns

# Looking at the first 5 colleges

In [4]:
  df.head()                                         #top 5 colleges in the dataframe

UNITID    OPEID  OPEID6                               INSTNM        CITY  \
0  100654   100200    1002             Alabama A & M University      Normal   
1  100663   105200    1052  University of Alabama at Birmingham  Birmingham   
2  100690  2503400   25034                   Amridge University  Montgomery   
3  100706   105500    1055  University of Alabama in Huntsville  Huntsville   
4  100724   100500    1005             Alabama State University  Montgomery   

  STABBR         ZIP  ACCREDAGENCY  INSTURL  NPCURL  ...  \
0     AL       35762           NaN      NaN     NaN  ...   
1     AL  35294-0110           NaN      NaN     NaN  ...   
2     AL  36117-3553           NaN      NaN     NaN  ...   
3     AL       35899           NaN      NaN     NaN  ...   
4     AL  36104-0271           NaN      NaN     NaN  ...   

   OMENRUP_PARTTIME_POOLED_SUPP  FTFTPCTPELL  FTFTPCTFLOAN   UG12MN   G12MN  \
0                           NaN       0.7106        0.7418   5207.0  1185.0   
1                           NaN       0.4009        0.5488  13308.0  9888.0   
2                           NaN       0.5000        0.5000    431.0   437.0   
3                           NaN       0.2012        0.4231   7519.0  2378.0   
4                           NaN       0.7594        0.7402   5229.0   687.0   

   SCUGFFN  POOLYRS_FTFTAIDPCT  FTFTPCTPELL_POOLED_SUPP  \
0   1410.0                 NaN                      NaN   
1   1948.0                 NaN                      NaN   
2      4.0                 NaN                      NaN   
3   1203.0                 NaN                      NaN   
4   1143.0                 NaN                      NaN   

   FTFTPCTFLOAN_POOLED_SUPP  SCUGFFN_POOLED  
0                       NaN             NaN  
1                       NaN             NaN  
2                       NaN             NaN  
3                       NaN             NaN  
4                       NaN             NaN  

[5 rows x 1986 columns]

# Looking at the last 5 colleges

In [7]:
df.tail()                                          #tail of the dataframe

UNITID    OPEID  OPEID6  \
7107  48312404   108163    1081   
7108  48387801  4223701   42237   
7109  48400201  4228101   42281   
7110  48778201  4221501   42215   
7111  49005401  4182601   41826   

                                                 INSTNM             CITY  \
7107                   Arizona State University at Yuma             Yuma   
7108  Bay Area Medical Academy - San Jose Satellite ...         San Jose   
7109           High Desert Medical College - Bakerfield      Bakersfield   
7110                                       BCI - Malden           Malden   
7111               HCI College - Fort Lauderdale Campus  Fort Lauderdale   

     STABBR        ZIP  ACCREDAGENCY  INSTURL  NPCURL  ...  \
7107     AZ  853656900           NaN      NaN     NaN  ...   
7108     CA      95113           NaN      NaN     NaN  ...   
7109     CA      93301           NaN      NaN     NaN  ...   
7110     MA  021480000           NaN      NaN     NaN  ...   
7111     FL      33309           NaN      NaN     NaN  ...   

      OMENRUP_PARTTIME_POOLED_SUPP  FTFTPCTPELL  FTFTPCTFLOAN  UG12MN  G12MN  \
7107                           NaN          NaN           NaN     NaN    NaN   
7108                           NaN          NaN           NaN     NaN    NaN   
7109                           NaN          NaN           NaN     NaN    NaN   
7110                           NaN          NaN           NaN     NaN    NaN   
7111                           NaN          NaN           NaN     NaN    NaN   

      SCUGFFN  POOLYRS_FTFTAIDPCT  FTFTPCTPELL_POOLED_SUPP  \
7107      NaN                 NaN                      NaN   
7108      NaN                 NaN                      NaN   
7109      NaN                 NaN                      NaN   
7110      NaN                 NaN                      NaN   
7111      NaN                 NaN                      NaN   

      FTFTPCTFLOAN_POOLED_SUPP  SCUGFFN_POOLED  
7107                       NaN             NaN  
7108                       NaN             NaN  
7109                       NaN             NaN  
7110                       NaN             NaN  
7111                       NaN             NaN  

[5 rows x 1986 columns]

# Looking at a random sample of 5 colleges

In [9]:
df.sample(5)                                         #Random 5 colleges

UNITID    OPEID  OPEID6                                        INSTNM  \
1325  157298   197400    1974  Louisville Presbyterian Theological Seminary   
234   110680   131700    1317            University of California-San Diego   
4934  441663  3642300   36423                       Institute of Technology   
6355  485342  4194200   41942               Aparicio-Levy Technical College   
577   128328   136900    1369               United States Air Force Academy   

              CITY STABBR         ZIP  ACCREDAGENCY  INSTURL  NPCURL  ...  \
1325    Louisville     KY  40205-1798           NaN      NaN     NaN  ...   
234       La Jolla     CA       92093           NaN      NaN     NaN  ...   
4934         Salem     OR       97305           NaN      NaN     NaN  ...   
6355         Tampa     FL       33610           NaN      NaN     NaN  ...   
577   USAF Academy     CO  80840-5002           NaN      NaN     NaN  ...   

      OMENRUP_PARTTIME_POOLED_SUPP  FTFTPCTPELL  FTFTPCTFLOAN   UG12MN  \
1325                           NaN          NaN           NaN      NaN   
234                            NaN       0.3074        0.3591  29613.0   
4934                           NaN       0.6846        0.8315    496.0   
6355                           NaN       0.3902        0.0000    270.0   
577                            NaN          NaN           NaN   4237.0   

       G12MN  SCUGFFN  POOLYRS_FTFTAIDPCT  FTFTPCTPELL_POOLED_SUPP  \
1325   173.0      NaN                 NaN                      NaN   
234   7172.0   5715.0                 NaN                      NaN   
4934     NaN    279.0                 NaN                      NaN   
6355     NaN     41.0                 NaN                      NaN   
577      NaN      NaN                 NaN                      NaN   

      FTFTPCTFLOAN_POOLED_SUPP  SCUGFFN_POOLED  
1325                       NaN             NaN  
234                        NaN             NaN  
4934                       NaN             NaN  
6355                       NaN             NaN  
577                        NaN             NaN  

[5 rows x 1986 columns]

# Creating a smaller data frame with the nine columns

In [10]:
df1=df[['UNITID','INSTNM','CITY','STABBR','ZIP','ADM_RATE','UGDS','TUITIONFEE_IN','MN_EARN_WNE_P10']]
df1                                                       #small dataframe

UNITID                                             INSTNM  \
0       100654                           Alabama A & M University   
1       100663                University of Alabama at Birmingham   
2       100690                                 Amridge University   
3       100706                University of Alabama in Huntsville   
4       100724                           Alabama State University   
...        ...                                                ...   
7107  48312404                   Arizona State University at Yuma   
7108  48387801  Bay Area Medical Academy - San Jose Satellite ...   
7109  48400201           High Desert Medical College - Bakerfield   
7110  48778201                                       BCI - Malden   
7111  49005401               HCI College - Fort Lauderdale Campus   

                 CITY STABBR         ZIP  ADM_RATE     UGDS  TUITIONFEE_IN  \
0              Normal     AL       35762    0.9027   4824.0         9857.0   
1          Birmingham     AL  35294-0110    0.9181  12866.0         8328.0   
2          Montgomery     AL  36117-3553       NaN    318.0         6900.0   
3          Huntsville     AL       35899    0.8123   6917.0        10280.0   
4          Montgomery     AL  36104-0271    0.9787   4189.0        11068.0   
...               ...    ...         ...       ...      ...            ...   
7107             Yuma     AZ   853656900       NaN      NaN            NaN   
7108         San Jose     CA       95113       NaN      NaN            NaN   
7109      Bakersfield     CA       93301       NaN      NaN            NaN   
7110           Malden     MA   021480000       NaN      NaN            NaN   
7111  Fort Lauderdale     FL       33309       NaN      NaN        22575.0   

      MN_EARN_WNE_P10  
0                 NaN  
1                 NaN  
2                 NaN  
3                 NaN  
4                 NaN  
...               ...  
7107              NaN  
7108              NaN  
7109              NaN  
7110              NaN  
7111              NaN  

[7112 rows x 9 columns]

UNITID       -               Unit ID for institution

INSTNM       -              Institution name

CITY         -                City

STABBR       -              State postcode

ZIP          -                 ZIP code

ADM_RATE     -                Rate of Admission

UGDS         -                number of certificate-seeking undergraduates

TUITIONFEE_IN     -           In-state tution fees

MN_EARN_WNE_P10   -           Mean aggregate of UG students who are enrolled 




# Look at the first 10 colleges

In [11]:
df1.head(10)                                    #head of the small dataframe

UNITID                               INSTNM            CITY STABBR  \
0  100654             Alabama A & M University          Normal     AL   
1  100663  University of Alabama at Birmingham      Birmingham     AL   
2  100690                   Amridge University      Montgomery     AL   
3  100706  University of Alabama in Huntsville      Huntsville     AL   
4  100724             Alabama State University      Montgomery     AL   
5  100751            The University of Alabama      Tuscaloosa     AL   
6  100760    Central Alabama Community College  Alexander City     AL   
7  100812              Athens State University          Athens     AL   
8  100830      Auburn University at Montgomery      Montgomery     AL   
9  100858                    Auburn University          Auburn     AL   

          ZIP  ADM_RATE     UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
0       35762    0.9027   4824.0         9857.0              NaN  
1  35294-0110    0.9181  12866.0         8328.0              NaN  
2  36117-3553       NaN    318.0         6900.0              NaN  
3       35899    0.8123   6917.0        10280.0              NaN  
4  36104-0271    0.9787   4189.0        11068.0              NaN  
5  35487-0100    0.5330  32387.0        10780.0              NaN  
6       35010       NaN   1404.0         4440.0              NaN  
7       35611       NaN   2801.0            NaN              NaN  
8  36117-3596    0.8240   4211.0         8020.0              NaN  
9       36849    0.8393  23391.0        10968.0              NaN

# Look at the last 10 colleges

In [12]:
df1.tail(10)                                             #Tail of the dataframe

UNITID                                             INSTNM  \
7102  48154401  National Personal Training Institute of Cleveland   
7103  48305401             Barber School of Pittsburgh - Ambridge   
7104  48312401                 Arizona State University at Tucson   
7105  48312402  Arizona State University - ASU Colleges at Lak...   
7106  48312403        Arizona State University at The Gila Valley   
7107  48312404                   Arizona State University at Yuma   
7108  48387801  Bay Area Medical Academy - San Jose Satellite ...   
7109  48400201           High Desert Medical College - Bakerfield   
7110  48778201                                       BCI - Malden   
7111  49005401               HCI College - Fort Lauderdale Campus   

                  CITY STABBR        ZIP  ADM_RATE  UGDS  TUITIONFEE_IN  \
7102  Highland Heights     OH      44143       NaN   NaN            NaN   
7103          Ambridge     PA  150030000       NaN   NaN            NaN   
7104            Tucson     AZ  857454284       NaN   NaN            NaN   
7105  Lake Havasu City     AZ  864036877       NaN   NaN         6904.0   
7106          Thatcher     AZ  855525545       NaN   NaN            NaN   
7107              Yuma     AZ  853656900       NaN   NaN            NaN   
7108          San Jose     CA      95113       NaN   NaN            NaN   
7109       Bakersfield     CA      93301       NaN   NaN            NaN   
7110            Malden     MA  021480000       NaN   NaN            NaN   
7111   Fort Lauderdale     FL      33309       NaN   NaN        22575.0   

      MN_EARN_WNE_P10  
7102              NaN  
7103              NaN  
7104              NaN  
7105              NaN  
7106              NaN  
7107              NaN  
7108              NaN  
7109              NaN  
7110              NaN  
7111              NaN

# Look at a random sample of 10 colleges

In [13]:
df1.sample(10)                                  #Random 10 colleges

UNITID                                             INSTNM  \
7104  48312401                 Arizona State University at Tucson   
6823  17296301    Anoka-Ramsey Community College-Cambridge Campus   
3941    242617  Inter American University of Puerto Rico-San G...   
5563    456588                      Salon Success Academy-Fontana   
5989    475699                    Bryant & Stratton College-Akron   
3235    218238                                  Limestone College   
662     132675                    North Florida Technical College   
769     137713                         Riveroak Technical College   
1556    166018                                  Hampshire College   
1177    153463                              American Hair Academy   

              CITY STABBR         ZIP  ADM_RATE    UGDS  TUITIONFEE_IN  \
7104        Tucson     AZ   857454284       NaN     NaN            NaN   
6823     Cambridge     MN   550085704       NaN     NaN         5059.0   
3941    San German     PR       00683    0.6899  4046.0         5898.0   
5563       Fontana     CA       92335       NaN    35.0            NaN   
5989        Copley     OH  44321-2798       NaN    88.0        17935.0   
3235       Gaffney     SC  29340-3799    0.5045  2491.0        25025.0   
662         Starke     FL       32091       NaN    90.0            NaN   
769       Live Oak     FL       32064       NaN   173.0            NaN   
1556       Amherst     MA  01002-5001    0.6282  1264.0        51608.0   
1177  Mt. Pleasant     IA       52641       NaN    24.0            NaN   

      MN_EARN_WNE_P10  
7104              NaN  
6823              NaN  
3941              NaN  
5563              NaN  
5989              NaN  
3235              NaN  
662               NaN  
769               NaN  
1556              NaN  
1177              NaN

# Finding and displaying UMBC

In [14]:
df1.loc[df1['INSTNM']=='University of Maryland-Baltimore County']                    

UNITID                                   INSTNM       CITY STABBR  \
1478  163268  University of Maryland-Baltimore County  Baltimore     MD   

        ZIP  ADM_RATE     UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
1478  21250    0.6021  11130.0        11518.0              NaN

# Finding Univ. of Maryland College Park

In [15]:
df1.loc[df1['INSTNM']=='University of Maryland-College Park']              

UNITID                               INSTNM          CITY STABBR    ZIP  \
1479  163286  University of Maryland-College Park  College Park     MD  20742   

      ADM_RATE     UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
1479     0.445  29273.0        10399.0              NaN

# Finding and displaying Loyola College Maryland

In [18]:
df1.loc[df1['INSTNM']=='Loyola University Maryland']

UNITID                      INSTNM       CITY STABBR         ZIP  \
1475  163046  Loyola University Maryland  Baltimore     MD  21210-2699   

      ADM_RATE    UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
1475    0.7528  3903.0        47560.0              NaN

# Finding and displaying Johns Hopskins

In [19]:
df1.loc[df1['INSTNM']=='Johns Hopkins University']

UNITID                    INSTNM       CITY STABBR         ZIP  \
1473  162928  Johns Hopkins University  Baltimore     MD  21218-2688   

      ADM_RATE    UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
1473    0.1254  5663.0        52170.0              NaN

# Calculate the min, max, and average in-state tution of the above four institutions.

In [23]:
colleges=['University of Maryland-Baltimore County','University of Maryland-College Park','Loyola University Maryland','Johns Hopkins University']

In [24]:
df2=df1.loc[df1['INSTNM'].isin(colleges)]

In [25]:
df2

UNITID                                   INSTNM          CITY STABBR  \
1473  162928                 Johns Hopkins University     Baltimore     MD   
1475  163046               Loyola University Maryland     Baltimore     MD   
1478  163268  University of Maryland-Baltimore County     Baltimore     MD   
1479  163286      University of Maryland-College Park  College Park     MD   

             ZIP  ADM_RATE     UGDS  TUITIONFEE_IN  MN_EARN_WNE_P10  
1473  21218-2688    0.1254   5663.0        52170.0              NaN  
1475  21210-2699    0.7528   3903.0        47560.0              NaN  
1478       21250    0.6021  11130.0        11518.0              NaN  
1479       20742    0.4450  29273.0        10399.0              NaN

In [26]:
df2['TUITIONFEE_IN'].min()

10399.0

In [27]:
df2['TUITIONFEE_IN'].max()

52170.0

In [29]:
df2['TUITIONFEE_IN'].mean()

30411.75

# Filtering the smaller dataset further to only keep colleges from Maryland, Virgina, Washington DC, Delaware, and Pennsylvinia. Saving the filtered dataset to a new file named "Five_States_Colleges.csv" for future analysis. 

In [30]:
states=['MD','VA','WA','DE','PA']

In [31]:
df4=df1.loc[df1['STABBR'].isin(states)]

In [32]:
df4

UNITID                                             INSTNM  \
65      102845                                    Charter College   
634     130837  Margaret H Rollins School of Nursing at Beebe ...   
635     130873                          Dawn Career Institute LLC   
636     130907         Delaware Technical Community College-Terry   
637     130934                          Delaware State University   
...        ...                                                ...   
7024  44378406                Strayer University-Allentown Campus   
7063  45794301   Somerset County Technology Center - Bedford Site   
7084  47503303      Relay Graduate School of Education - Delaware   
7089  47503308  Relay Graduate School of Education - Philadelp...   
7103  48305401             Barber School of Pittsburgh - Ambridge   

               CITY STABBR         ZIP  ADM_RATE     UGDS  TUITIONFEE_IN  \
65        Vancouver     WA  98683-7575       NaN   2640.0        21908.0   
634           Lewes     DE       19958       NaN     59.0            NaN   
635          Newark     DE       19702       NaN    108.0            NaN   
636           Dover     DE       19901       NaN  12408.0         4848.0   
637           Dover     DE       19901    0.4518   3678.0         7868.0   
...             ...    ...         ...       ...      ...            ...   
7024  Center Valley     PA       18034       NaN      NaN        13857.0   
7063        Bedford     PA       15522       NaN      NaN            NaN   
7084     Wilmington     DE       19801       NaN      NaN            NaN   
7089   Philadelphia     PA   191224906       NaN      NaN            NaN   
7103       Ambridge     PA   150030000       NaN      NaN            NaN   

      MN_EARN_WNE_P10  
65                NaN  
634               NaN  
635               NaN  
636               NaN  
637               NaN  
...               ...  
7024              NaN  
7063              NaN  
7084              NaN  
7089              NaN  
7103              NaN  

[785 rows x 9 columns]

In [34]:
df4.to_csv('Five_States_Colleges.csv', index=False)     